### PR TITLE
theme.json docs: clarify naming schema for CSS Custom Properties

### PR DESCRIPTION
**Pull request by oandregal**
This PR updates the theme.json docs to include more information about the naming schema of CSS Custom Properties generated by the system.

View [docs live](https://github.com/WordPress/gutenberg/blob/120bd1c1c32fff2a76d9fa255e5dd20b17e4d687/docs/designers-developers/developers/themes/theme-json.md).

### CODE REVIEW

File: docs/designers-developers/developers/themes/theme-json.md [hunk 1]
```diff
@@ -2,35 +2,113 @@
...
+
+### CSS Custom Properties
+
+There are some areas of styling that would benefit from having shared values that can change across a site instantly.
+
+To address this need, we've started to experiment with CSS Custom Properties, aka CSS Variables, in some places:
+
+- **Presets**: [color palettes](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-color-palettes), [font sizes](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes), or [gradients](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-gradient-presets) declared by the theme are converted to CSS Custom Properties and enqueued both the front-end and the editors.
+
+{% codetabs %}
```
Comment (nosolosw) [inline docs/designers-developers/developers/themes/theme-json.md:53]: I'm not sure this `codetabs` thing is going to work the way I expected it to work. This has been used to show ES5/ESNext code examples side by side. Can we use it for other things as well?

I thought I'd figure it out by pushing the changes to the handbook and see.

Comment (mkaz) [reply docs/designers-developers/developers/themes/theme-json.md:53]: Clever use, I think it will work. :+1:

**Review by mkaz — APPROVED** [review]
This makes sense to me and offers clarification and guidance around naming conventions.
I don't have any suggestions myself, so will approve but will leave open if @jasmussen wants to push anything up prior to merge. :+1:

**Review by jasmussen — APPROVED** [review]
Thank you, so helpful! I think this is the key bit:

```
**Presets** such as `--wp--preset--color--black` can be divided into the following chunks:

- `--wp`: prefix to namespace the CSS variable.
- `preset `: indicates is a CSS variable that belongs to the presets.
- `color`: indicates which preset category the variable belongs to. It can be `color`, `font-size`, `gradients`.
- `black`: the `slug` of the particular preset value.
```

I think this structure makes a lot of sense. In most cases, I'd prefer a prefix, and a very short variable name. But for global styles, and because of how it works, I understand it's important with a structured system for these, in addition to the preset. So this seems good.

Question, though, what does the `--` mean instead of just `-`? I'm guessing it's a BEM-like category separator, whereas a single dash is just meant as a word-hyphen?

@youknowriad is AFK at the moment, but just wanted to leave a ping in his inbox for when he's back!

Thank you André!

**Comment by nosolosw** [comment]
> Question, though, what does the -- mean instead of just -? I'm guessing it's a BEM-like category separator, whereas a single dash is just meant as a word-hyphen?

Hadn't thought about it in terms of its similarity to BEM naming, but it's definitely similar: it separates "categories".

The rationale for using `--` as a separator between categories is described a bit below in the doc and has two functions:     

- Readibility, for human understanding.
- Parseability, for machine understanding. Using a defined structure allows machines to understand the meaning of the property `--wp--preset--color--black`: it's a value bounded to the color preset whose slug is "black", which then gives us room to do more things with them.

**Comment by nosolosw** [comment]
ok, I've added a new section called "why using `--` as a separator" and made a few other tweaks to link it to BEM categories: see [docs live](https://github.com/WordPress/gutenberg/blob/ff3b7f8f4cb8c458222b0c7253fa6c03aba9ff2f/docs/designers-developers/developers/themes/theme-json.md#why-using----as-a-separator).

File: docs/designers-developers/developers/themes/theme-json.md (hunk 1 above)
Comment (nosolosw) [reply docs/designers-developers/developers/themes/theme-json.md:54]: It didn't go very well :( I've created https://github.com/WordPress/gutenberg/pull/28655 to see if I can fix it. Comparing it to other `codetabs` uses, I don't see any missing markup.

Comment (nosolosw) [reply docs/designers-developers/developers/themes/theme-json.md:54]: Using single words fixed it :muscle:

**Pull request merged**